crates/ruma-common/src/push/condition.rs

```diff
@@ -197,6 +211,35 @@ impl PushCondition {
                 .get(key)
                 .and_then(FlattenedJsonValue::as_array)
                 .is_some_and(|a| a.contains(value)),
+            #[cfg(feature = "unstable-msc4306")]
+            Self::ThreadSubscription { subscribed: must_be_subscribed } => {
+                let Some(has_thread_subscription_fn) = &context.has_thread_subscription_fn else {
+                    // If we don't have a function to check thread subscriptions, we can't
+                    // determine if the condition applies.
+                    return false;
+                };
+
+                // The event must have a relation of type `m.thread`.
+                if event.get_str("content.m\\.relates_to.rel_type") != Some("m.thread") {
+                    return false;
+                }
+
+                // Retrieve the thread root event ID.
+                let Some(Ok(thread_root)) = event
+                    .get_str("content.m\\.relates_to.event_id")
+                    .map(|str| str.parse::<OwnedEventId>())
```

Comment:
We can avoid an allocation here:

```suggestion
                    .map(|str| <&EventId>::try_from(str))
```